course/include/isometry/isometry.hpp

```diff
@@ -6,19 +6,78 @@
 
 #pragma once
 
+#include <algorithm>
 #include <cmath>
 #include <initializer_list>
 #include <iostream>
+#include <limits>
 #include <sstream>
+#include <stdexcept>
 #include <string>
+#include <type_traits>
 
 namespace ekumen {
 
 namespace math {
 
-/*
- * your code here
- */
+template <class T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, bool>::type
+almost_equal(T x, T y, int ulp) {
+  return std::fabs(x - y) <=
+             std::numeric_limits<T>::epsilon() * std::fabs(x + y) * ulp ||
+         std::fabs(x - y) < std::numeric_limits<T>::min();
+}
+
+class Vector3 {
+ public:
+  Vector3(double x, double y, double z) : x_{x}, y_{y}, z_{z} {};
+  Vector3(std::initializer_list<double> list);
+  Vector3() : x_{0}, y_{0}, z_{0} {};
+
+  double operator[](const int i) const;
+  double& operator[](const int i);
+
+  double x() const { return x_; }
+  double& x() { return x_; }
+  double y() const { return y_; }
+  double& y() { return y_; }
+  double z() const { return z_; }
+  double& z() { return z_; }
+
+  double dot(const Vector3& r_vector) const;
+  double norm() const;
+  Vector3 cross(const Vector3& r_vector) const;
+
+  bool operator==(const Vector3& r_vector) const;
+  bool operator!=(const Vector3& r_vector) const;
+  Vector3& operator-=(const Vector3& r_vector);
+  Vector3 operator-(const Vector3& r_vector) const;
+  Vector3& operator+=(const Vector3& r_vector);
+  Vector3 operator+(const Vector3& r_vector) const;
+  Vector3& operator*=(const double& int_value);
+  Vector3 operator*(const double& int_value) const;
```

Comment:
Please rename `int_value` to just `value` since it's not an int.

Reply:
In general, avoid declaring the name of the type of the variable in the name. Lends itself for problems when you refactor the code.